### ISSUE TITLE
Steam (runtime) has no network; cannot update client or login

### ISSUE THREAD

**Issue by mlandalv** [ValveSoftware/steam-for-linux#7576]
### System information

* Steam version: [steam-1.0.0.68-1](https://archlinux.org/packages/multilib/x86_64/steam/)
* Distro: Arch Linux (5.9.14-arch1-1)
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
It all started when I was trying some games through Proton (Killing Floor 2 and GTA V), but I think the real problem is the Steam Client Runtime. Both games seemed to "start", but none of them had a network connection. Perhaps same issue as [this](https://github.com/ValveSoftware/Proton/issues/162) where someone suggests that **libgnutls** fails to load which results in failed network attempts.

I searched around and someone suggested to use steam runtime instead of steam native which I was using, but same problem, still no network in Proton games.

Since I have the games on a separate drive I figured I might as well try a fresh install of Steam. So I deleted `~/.local/share/Steam` and `~/.steam`. Uninstalled [`steam-native-runtime`](https://archlinux.org/packages/multilib/x86_64/steam-native-runtime/) and all of the dependencies that came along.

Now when I run the client it only shows a modal "Steam - Fatal Error": "Fatal Error: Steam needs to be online to update. Please confirm your network connection and try again."
![image](https://user-images.githubusercontent.com/710584/103274723-9fe4ff00-49c2-11eb-81b9-e0d44fc15bb2.png)

Since Steam (runtime) worked when I had the `steam-native-runtime` installed, I assume some of its dependencies perhaps are required and should be bundled with the runtime? But I can't figure out which one. I'm guessing that if I get Steam Runtime to work, then perhaps the networking in Proton games will start working too.

<details>
<summary>List of my installed multilib packages</summary>
<div>

```lib32-alsa-lib
lib32-alsa-plugins
lib32-attr
lib32-bzip2
lib32-curl
lib32-e2fsprogs
lib32-expat
lib32-icu
lib32-keyutils
lib32-krb5
lib32-libdrm
lib32-libelf
lib32-libffi
lib32-libglvnd
lib32-libgpg-error
lib32-libidn
lib32-libidn2
lib32-libldap
lib32-libpciaccess
lib32-libpsl
lib32-libssh2
lib32-libtasn1
lib32-libunistring
lib32-libunwind
lib32-libx11
lib32-libxau
lib32-libxcb
lib32-libxcrypt
lib32-libxdamage
lib32-libxdmcp
lib32-libxext
lib32-libxfixes
lib32-libxml2
lib32-libxshmfence
lib32-libxss
lib32-libxxf86vm
lib32-llvm-libs
lib32-lm_sensors
lib32-lz4
lib32-mesa
lib32-ncurses
lib32-nspr
lib32-nss
lib32-nvidia-utils
lib32-openssl
lib32-p11-kit
lib32-readline
lib32-sqlite
lib32-util-linux
lib32-vulkan-icd-loader
lib32-wayland
lib32-xz
lib32-zlib
lib32-zstd
steam
```
</div>
</details>

### Steps for reproducing this issue

1. Uninstall `steam-native-runtime`
2. `~/.local/share/Steam` and `~/.steam`
3. Run `steam`

### What I've tried

So basically I've been trying to figure out if there is any `so` file missing.

#### Running `steam` and trying to figure out if something obvious is missing. No luck.

<details>
<summary>See output</summary
<div>

```
Running Steam on arch rolling 64-bit
STEAM_RUNTIME is enabled automatically
Pins up-to-date!
Can't find 'steam-runtime-check-requirements', continuing anyway
/home/martin/.local/share/Steam/ubuntu12_32/steam
[2020-12-29 10:56:23] Startup - updater built Oct 28 2020 23:34:22
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
Installing breakpad exception handler for appid(steam)/version(0)
[2020-12-29 10:56:24] Loading cached metrics from disk (/home/martin/.local/share/Steam/package/steam_client_metrics.bin)
[2020-12-29 10:56:24] Failed to load cached hosts file (File 'update_hosts_cached.vdf' not found), using defaults
[2020-12-29 10:56:24] Using the following download hosts for Public, Realm steamglobal
[2020-12-29 10:56:24] 1. http://client-download.steampowered.com, /client/, Realm 'steamglobal', weight was 100, source = 'baked in'
[2020-12-29 10:56:24] 2. http://media.steampowered.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
Installing breakpad exception handler for appid(steam)/version(1.0)
[2020-12-29 10:56:24] Verifying installation...
[2020-12-29 10:56:24] Unable to read and verify install manifest /home/martin/.local/share/Steam/package/steam_client_ubuntu12.installed
[2020-12-29 10:56:24] Verification complete
[2020-12-29 10:56:24] Downloading Update...
[2020-12-29 10:56:24] Checking for available update...
[2020-12-29 10:56:24] Downloading manifest: client-download.steampowered.com/client/steam_client_ubuntu12
Installing breakpad exception handler for appid(steam)/version(1.0)
[2020-12-29 10:56:24] Download failed: http error 0 (client-download.steampowered.com/client/steam_client_ubuntu12)
[2020-12-29 10:56:24] Downloading manifest: media.steampowered.com/client/steam_client_ubuntu12
[2020-12-29 10:56:24] Download failed: http error 0 (media.steampowered.com/client/steam_client_ubuntu12)
[2020-12-29 10:56:24] DownloadManifest - exhausted list of download hosts
[2020-12-29 10:56:24] failed to load manifest from buffer.
[2020-12-29 10:56:24] Failed to load manifest
[2020-12-29 10:56:24] Error: Download failed: http error 0
[2020-12-29 10:56:24] Saving metrics to disk (/home/martin/.local/share/Steam/package/steam_client_metrics.bin)
[2020-12-29 10:56:24] Error: Steam needs to be online to update.	 Please confirm your network connection and try again.
[2020-12-29 10:56:25] Shutdown
/data/src/tier0/threadtools.cpp (3658) : Assertion Failed: Illegal termination of worker thread 'Thread(0x0x58916ef0/0x0xf469ca'
/data/src/tier0/threadtools.cpp (3658) : Assertion Failed: Illegal termination of worker thread 'Thread(0x0x58916ef0/0x0xf469ca'
Installing breakpad exception handler for appid(steam)/version(1.0)
crash_20201229105626_5.dmp[15004]: Uploading dump (out-of-process)
/tmp/dumps/crash_20201229105626_5.dmp
crash_20201229105626_5.dmp[15004]: Finished uploading minidump (out-of-process): success = no
crash_20201229105626_5.dmp[15004]: error: Couldn't resolve host name
crash_20201229105626_5.dmp[15004]: file ''/tmp/dumps/crash_20201229105626_5.dmp'', upload no: ''Couldn't resolve host name''
```
</div>
</details>

[crash_20201229105626_5.dmp.zip](https://github.com/ValveSoftware/steam-for-linux/files/5750319/crash_20201229105626_5.dmp.zip)

#### Find missing runtime libraries

(As suggested in the [Arch Wiki](https://wiki.archlinux.org/index.php/Steam/Troubleshooting#Debugging_shared_libraries).)

```
cd ~/.steam/root/ubuntu12_32
file * | grep ELF | cut -d: -f1 | LD_LIBRARY_PATH=. xargs ldd | grep 'not found' | sort | uniq
[Empty result]
```

#### Running `env DEBUGGER=gdb steam`, then typing `run`
Since I have no idea how to use gdb that didn't give anything new.
<details>
<summary>See output</summary>
<div>

```(gdb) run
Starting program: /home/martin/.local/share/Steam/ubuntu12_32/steam 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[2020-12-29 11:02:58] Startup - updater built Oct 28 2020 23:34:22
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
Installing breakpad exception handler for appid(steam)/version(0)
[2020-12-29 11:02:58] Loading cached metrics from disk (/home/martin/.local/share/Steam/package/steam_client_metrics.bin)
[2020-12-29 11:02:58] Failed to load cached hosts file (File 'update_hosts_cached.vdf' not found), using defaults
[2020-12-29 11:02:58] Using the following download hosts for Public, Realm steamglobal
[2020-12-29 11:02:58] 1. http://client-download.steampowered.com, /client/, Realm 'steamglobal', weight was 100, source = 'baked in'
[2020-12-29 11:02:58] 2. http://media.steampowered.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
[New Thread 0xf46d0ac0 (LWP 15563)]
Installing breakpad exception handler for appid(steam)/version(1.0)
[2020-12-29 11:02:58] Verifying installation...
[2020-12-29 11:02:58] Unable to read and verify install manifest /home/martin/.local/share/Steam/package/steam_client_ubuntu12.installed
[2020-12-29 11:02:58] Verification complete
[2020-12-29 11:02:58] Downloading Update...
[2020-12-29 11:02:58] Checking for available update...
[2020-12-29 11:02:58] Downloading manifest: client-download.steampowered.com/client/steam_client_ubuntu12
[New Thread 0xf43ffac0 (LWP 15564)]
Installing breakpad exception handler for appid(steam)/version(1.0)
[2020-12-29 11:02:58] Download failed: http error 0 (client-download.steampowered.com/client/steam_client_ubuntu12)
[2020-12-29 11:02:58] Downloading manifest: media.steampowered.com/client/steam_client_ubuntu12
[2020-12-29 11:02:58] Download failed: http error 0 (media.steampowered.com/client/steam_client_ubuntu12)
[2020-12-29 11:02:58] DownloadManifest - exhausted list of download hosts
[2020-12-29 11:02:58] failed to load manifest from buffer.
[2020-12-29 11:02:58] Failed to load manifest
[2020-12-29 11:02:58] Error: Download failed: http error 0
[2020-12-29 11:02:58] Saving metrics to disk (/home/martin/.local/share/Steam/package/steam_client_metrics.bin)
[2020-12-29 11:02:58] Error: Steam needs to be online to update.	 Please confirm your network connection and try again.
[Thread 0xf46d0ac0 (LWP 15563) exited]
[2020-12-29 11:03:22] Shutdown
[Thread 0xf43ffac0 (LWP 15564) exited]
/data/src/tier0/threadtools.cpp (3658) : Assertion Failed: Illegal termination of worker thread 'Thread(0x0x56a1aef0/0x0xf46d0a'
/data/src/tier0/threadtools.cpp (3658) : Assertion Failed: Illegal termination of worker thread 'Thread(0x0x56a1aef0/0x0xf46d0a'
/data/src/tier0/threadtools.cpp 3658 Assertion Failed: Illegal termination of worker thread 'Thread(0x0x56a1aef0/0x0xf46d0a'
Thread 1 "steam" received signal SIGTRAP, Trace/breakpoint trap.
```
</div>
</details>

#### I've tried the `-tcp` switch (#7232)

Same error; still no network.

#### Installing `steam-native-runtime`

Networking in the client and native game now works but networking in Proton doesn't.

[Gist with my System Information](https://gist.github.com/mlandalv/94a1628a848d6580976de83c0782d24b)

**Comment by mlandalv** [comment]
I figured it out, see [this](https://bbs.archlinux.org/viewtopic.php?id=262187) thread over at the Arch Linux forum.

TL;DR;
When using **systemd-networkd** and **systemd-resolved** I had to install **lib32-systemd** for the Steam client to work. Still no network in Proton 5.13.
When I changed to **dhcpcd**, the network worked in both the Steam Client *and* Proton 5.13.

This issue can be closed.

**Comment by ernstp** [comment]
This looks exactly like my issue on Ubuntu. However installing libsystemd0:i386 did not solve it. What did you do afterwards exactly... ?

**Comment by mlandalv** [comment]
I switched to [dhcpcd](https://wiki.archlinux.org/index.php/Dhcpcd) instead. Fixed the network issues with Steam _and_ proton.

**Comment by ernstp** [comment]
I solved my issue now, by installing nscd!
And yeah, I had the exact same thing with Steam _and_ proton.
I have no idea why it helped, all getent hosts results look the same etc...
